features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/crypto/sendfailure/resolve/ResolveVerifiedUserSendFailureView.kt

```diff
@@ -0,0 +1,156 @@
+
+    fun dismiss() {
+        state.eventSink(ResolveVerifiedUserSendFailureEvents.Dismiss)
+    }
+
+    fun onRetryClick() {
+        state.eventSink(ResolveVerifiedUserSendFailureEvents.Retry)
+    }
+
+    fun onResolveAndResendClick() {
+        state.eventSink(ResolveVerifiedUserSendFailureEvents.ResolveAndResend)
+    }
+
+    LaunchedEffect(state.verifiedUserSendFailure) {
+        if (state.verifiedUserSendFailure is VerifiedUserSendFailure.None) {
+            sheetState.hide()
+            showSheet = false
+        } else {
+            showSheet = true
+        }
+    }
+
+    Box(modifier = modifier) {
+        if (showSheet) {
+            ModalBottomSheet(
+                modifier = Modifier
+                    .systemBarsPadding()
+                    .navigationBarsPadding(),
+                sheetState = sheetState,
+                onDismissRequest = ::dismiss,
+            ) {
+                IconTitleSubtitleMolecule(
+                    modifier = Modifier.padding(24.dp),
+                    title = state.verifiedUserSendFailure.title(),
+                    subTitle = state.verifiedUserSendFailure.subtitle(),
+                    iconImageVector = CompoundIcons.Error(),
+                    iconTint = ElementTheme.colors.iconCriticalPrimary,
+                    iconBackgroundTint = ElementTheme.colors.bgCriticalSubtle,
+                )
+                ButtonColumnMolecule(
+                    modifier = Modifier.padding(horizontal = 16.dp, vertical = 16.dp),
+                ) {
+                    Button(
+                        modifier = Modifier.fillMaxWidth(),
+                        text = state.verifiedUserSendFailure.resolveAction(),
+                        showProgress = state.resolveAction.isLoading(),
+                        onClick = ::onResolveAndResendClick
+                    )
+                    OutlinedButton(
+                        modifier = Modifier.fillMaxWidth(),
+                        text = stringResource(id = CommonStrings.action_retry),
+                        showProgress = state.retryAction.isLoading(),
+                        onClick = ::onRetryClick
+                    )
+                    TextButton(
+                        modifier = Modifier.fillMaxWidth(),
+                        text = stringResource(id = CommonStrings.action_cancel_for_now),
+                        onClick = ::dismiss,
+                    )
+                }
+            }
+        }
+    }
+}
+
+@Composable
+private fun VerifiedUserSendFailure.title(): String {
+    return when (this) {
+        is VerifiedUserSendFailure.UnsignedDevice -> stringResource(id = CommonStrings.screen_resolve_send_failure_unsigned_device_title, userDisplayName)
+        is VerifiedUserSendFailure.ChangedIdentity -> stringResource(
+            id = CommonStrings.screen_resolve_send_failure_changed_identity_title,
+            userDisplayName
+        )
+        VerifiedUserSendFailure.None -> ""
```

Comment:
Maybe these should be `error("This method should never be called for this state")`? Up to you, it would allow us to catch any regressions faster (and maybe make some users angry 😅 ).